docs/softwaredevelopment.md

```diff
@@ -1 +1,124 @@
-# Software development
\ No newline at end of file
+# Software development
+
+## How to name software products
+Read this guide when you need to name components for your software product, for example applications, software libraries, plugins or frameworks. Your users should understand what something does from its name.
+
+### User needs
+Users should to be able to infer a basic understanding of what a given thing does from its name.
+
+### Your product name should be self-descriptive
+Names should be self-descriptive – you should be able to infer a basic understanding of what something does from its name. 
+
+Avoid naming things using puns, uncommon acronyms, version numbers, or with a brand. All of these make it difficult for others to understand what the thing does. Puns and acronyms (unless they’re very widely understood outside government, like API) probably won’t be understood by anyone coming to your thing later. Version numbers and brand names (including names of parts of government) are likely to change over time as the thing adapts to changing user needs and policies.
+
+If your service is an instance of a well-known type of system (like an intranet or help desk), it is appropriate to include the name of the government body it’s associated with so people can understand the context it applies to. Where possible, avoid exposing that name to people (usually citizens or organisations outside government) who shouldn’t have to understand how government is structured to do what they’re trying to do.
+
+Ensure you use the same name consistently whenever you’re referring to the same thing. For example, in most cases the name of an application’s git repository should match its hostname. Similarly, the team responsible for an application should probably be called something related to the application and its git repositories.
+Make sure you use the same name consistently whenever you’re referring to the same product. 
+
+### Other resources
+The [GOV.UK Service Manual](https://www.gov.uk/service-manual) has good guidance for [naming services](https://www.gov.uk/service-manual/design/naming-your-service), a lot of which will also be relevant when naming applications or packages.
+
+## Programming languages
+
+We use a range of programming languages at DDS because we think using the right tool for the job gives us the best chance of building services that best meet users’ needs. This document does not apply to choosing ‘off the shelf’ software (open source or not).
+
+We focus on using a small number of programming languages for core software development tasks.
+
+This should make it easier for developers to:
+
+- move around the organisation
+- develop shared components
+- improve their personal development
+- master how they operate applications
+
+### Frontend development
```

Comment:
For consistency with the next section, should we be a bit more explicit about what we're using here (e.g. JS / React), then linking to the Service Manual as well?